SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -199,15 +199,33 @@ void ActorSystem::ProcessAdds(const FEntitySubViewUpdate& SubViewUpdate)
 			{
 				RefreshEntity(Delta.EntityId);
 			}
+		}
+	}
+}
 
-			if (SubViewUpdate.SubViewType != ENetRole::ROLE_SimulatedProxy)
+void ActorSystem::ProcessAuthorityGains(const FEntitySubViewUpdate& SubViewUpdate)
+{
+	for (const EntityDelta& Delta : SubViewUpdate.EntityDeltas)
+	{
+		if ((Delta.Type == EntityDelta::ADD || Delta.Type == EntityDelta::TEMPORARILY_REMOVED)
+			&& SubViewUpdate.SubViewType != ENetRole::ROLE_SimulatedProxy)
+		{
+			const Worker_EntityId EntityId = Delta.EntityId;
+			if (SubViewUpdate.SubViewType == ENetRole::ROLE_Authority)
```

Comment:
Can reduce the nesting with an &&

Reply:
Also can this condition ever fail?

Reply:
if you mean the Role_Auth one, then would fail on ownership subview

If you mean the HasEntityBeenRequestedForDelete, then possibly as a result of an actor being killed during some repnotify

Reply:
I'm pretty sure this is only using the unreal role as a hack and that autonomous can't be used. Which means we should really just use our own enum probably. Or just split the functions based on which subview they are expecting

Reply:
oh right that's the unreal enum, whoops

Reply:
Ah so SubViewUpdate.SubViewType won't be autonomous? will remove the check

Reply:
actually does seem like autonomousproxy will be used from the creation of the things

```
	const FEntitySubView SubViews[]{
		{ AuthoritySubView, ENetRole::ROLE_Authority },
		{ OwnershipSubView, ENetRole::ROLE_AutonomousProxy },
		{ SimulatedSubView, ENetRole::ROLE_SimulatedProxy },
	};
	```